MonoGame.Framework/Color.cs

```diff
@@ -273,14 +288,43 @@ public Color(int r, int g, int b)
         public Color(int r, int g, int b, int alpha)
         {
             _packedValue = 0;
-            R = (byte)MathHelper.Clamp(r, Byte.MinValue, Byte.MaxValue);
-            G = (byte)MathHelper.Clamp(g, Byte.MinValue, Byte.MaxValue);
-            B = (byte)MathHelper.Clamp(b, Byte.MinValue, Byte.MaxValue);
-            A = (byte)MathHelper.Clamp(alpha, Byte.MinValue, Byte.MaxValue);
+            if (((r | g | b | alpha) & 0xFFFFFF00) != 0)
+            {
+                R = (byte)MathHelper.Clamp(r, Byte.MinValue, Byte.MaxValue);
+                G = (byte)MathHelper.Clamp(g, Byte.MinValue, Byte.MaxValue);
+                B = (byte)MathHelper.Clamp(b, Byte.MinValue, Byte.MaxValue);
+                A = (byte)MathHelper.Clamp(alpha, Byte.MinValue, Byte.MaxValue);
+            }
+            else
+            {
+                R = (byte)r;
+                G = (byte)g;
+                B = (byte)b;
+                A = (byte)alpha;
+            }
+        }
+
+        /// <summary>
+        /// Constructs an RGBA color from scalars representing red, green, blue and alpha values.
+        /// </summary>
+        /// <remarks>
+        /// This overload sets the values directly without clamping, and may therefore be faster than the other overloads.
+        /// </remarks>
+        /// <param name="r"></param>
+        /// <param name="g"></param>
+        /// <param name="b"></param>
+        /// <param name="alpha"></param>
+        public Color(byte r, byte g, byte b, byte alpha)

```

Comment:
The `Color.FromBytes` version has the advantage that it will never fall back to the `int` implementation, but rather raises a compiler error. On the other hand, it adds additional API surface, and with the other optimizations here the performance advantage might actually be quite small.